shared/src/main/scala/squants/Quantity.scala

```diff
@@ -176,7 +178,9 @@ abstract class Quantity[A <: Quantity[A]] extends Serializable with Ordered[A] {
    *
    * @return
    */
-  override def hashCode() = toString.hashCode()
+  override def hashCode() = {
+    Objects.hash(dimension, Double.box(to(dimension.primaryUnit)))
```

Comment:
Unfortunately, the resulting hashCode is not stable across JVM runs because the `dimension` itself may not have a stable `hashCode`.